src/core/sig_state.ml

```diff
@@ -24,17 +24,62 @@ type sig_state =
 
 type t = sig_state
 
-(** [init_with sign] creates a state from the signature [sign] with no symbols
-   in scope, module aliases, builtins or printing hints. *)
-let of_sign : Sign.t -> sig_state = fun sign ->
-  { signature = sign
-  ; in_scope  = StrMap.empty
-  ; aliases   = StrMap.empty
-  ; builtins  = StrMap.empty
-  ; pp_hints  = SymMap.empty }
+(** Symbols and signature used for unification rules. *)
+module Unif_rule =
+  struct
 
-(** [empty] state. *)
-let empty = of_sign (Sign.create [])
+    (** Ghost signature holding the symbols used in unification rules. This
+        signature is an automatic dependency of all other signatures, and is
+        automatically loaded. *)
+    let sign : Sign.t =
+      let open Files in
+      let pth = Path.ghost "unif_rule" in
+      let s = Sign.create pth in
+      (* Remove the dependency on itself. *)
+      s.sign_deps := PathMap.remove pth !(s.sign_deps);
+      Sign.loaded := Files.PathMap.add pth s !(Sign.loaded);
+      s
+
+    (** Symbol representing an atomic unification problem. The term [equiv t
+        u] represents [t ≡ u]. The left-hand side of a unification rule is
+        made of only one unification. *)
+    let equiv : sym =
+      Sign.add_symbol sign Public Defin (Pos.none "#equiv") Kind []
+
+    (** Holds a list of equivalences. The right-hand side of a unification
+        rule is made of such a list, [list (equiv t u) (equiv v w) ...]. *)
+    let comma : sym =
+      Sign.add_symbol sign Public Defin (Pos.none "#comma") Kind []
+
+    (** Pretty printing hints for symbols [equiv] and [comma]. *)
+    let pp_hints : pp_hint SymMap.t =
+      let open Stdlib in
+      let res = ref SymMap.empty in
+      res := begin
+        let h = Infix("≡", Assoc_none, 1.1, Pos.none ([], "#equiv")) in
```

Comment:
[] should be replaced by pth.

Reply:
You need to move pth to make it an element of the module (and rename it into "path")